### PR TITLE
Point to S3 DefaultSprites

### PR DESCRIPTION
**Pull request by epeach**
Reverts code-dot-org/code-dot-org#42826

This is the final part of the change that moves from using a local defaultSprites.json to loading one on S3. The challenge is that this turns a synchronous operation into an asynchronous operation.

We've reverted this a few times as we discover our previous solution wasn't adequately loading the sprites before the page was rendered.

To minimize scope, I broke out parts of this work into smaller PRs, referenced below. This is the last part of this work.

**Investigation for cause of latest revert:**
We are hitting this https://github.com/code-dot-org/code-dot-org/blob/b27ef171412f2da5658f851f249de77b055ac54c/apps/src/p5lab/spritelab/blocks.js#L44

This traces back to this not being called before we hit that^. https://github.com/code-dot-org/code-dot-org/blob/ae0430a2643de2af1e11c2191e80319f9f9c204d/apps/src/p5lab/redux/animationList.js#L265

Which traces back to the init in P5Lab where we load the animations but don't pause initializing the rest of it while we wait to load the animations.

**Fix:**
I pulled the 'load default sprites' call into loadSpriteLab file and then wait for that to complete before passing it through to each LabType. If the request fails, we load a paired down version of the defaultSprites list (just two sprites) and log an error so we can see how often this happens. Loading this paired down, back-up list of sprites allows us to load the page and not block student learning, while mainly relying on the s3 version of default sprites.
Slack discussion where we decided to move forward with the 'back-up list' approach: https://codedotorg.slack.com/archives/C1B3PNDL7/p1645751499386679

### CODE REVIEW

File: apps/src/sites/studio/pages/init/loadGamelab.js [hunk 1]
```diff
@@ -3,17 +3,22 @@ import {singleton as studioApp} from '@cdo/apps/StudioApp';
 import GameLab from '@cdo/apps/p5lab/gamelab/GameLab';
 import skins from '@cdo/apps/p5lab/skins';
 import levels from '@cdo/apps/p5lab/levels';
+import {getDefaultListMetadata} from '@cdo/apps/assetManagement/animationLibraryApi';
 
 export default function loadGamelab(options) {
   options.skinsModule = skins;
-  var gamelab = new GameLab();
+  return getDefaultListMetadata().then(defaultSprites => {
+    var gamelab = new GameLab(defaultSprites);
```
Comment (maddiedierker) [inline apps/src/sites/studio/pages/init/loadGamelab.js]: nit: we should use let or const here to benefit from their scoping differences from var

File: apps/src/sites/studio/pages/init/loadSpritelab.js [hunk 2]
```diff
@@ -4,22 +4,25 @@ import SpriteLab from '@cdo/apps/p5lab/spritelab/SpriteLab';
...
 import skins from '@cdo/apps/p5lab/skins';
 import levels from '@cdo/apps/p5lab/levels';
+import {getDefaultListMetadata} from '@cdo/apps/assetManagement/animationLibraryApi';
 
 export default function loadSpritelab(options) {
   options.skinsModule = skins;
   options.blocksModule = blocks;
-  var spritelab = new SpriteLab();
+  return getDefaultListMetadata().then(defaultSprites => {
+    var spritelab = new SpriteLab(defaultSprites);
```
Comment (maddiedierker) [inline apps/src/sites/studio/pages/init/loadSpritelab.js]: nit: we should use let or const here to benefit from their scoping differences from var

File: apps/src/sites/studio/pages/init/loadGamelab.js [hunk 3]
```diff
@@ -3,17 +3,22 @@ import {singleton as studioApp} from '@cdo/apps/StudioApp';
 import GameLab from '@cdo/apps/p5lab/gamelab/GameLab';
 import skins from '@cdo/apps/p5lab/skins';
 import levels from '@cdo/apps/p5lab/levels';
+import {getDefaultListMetadata} from '@cdo/apps/assetManagement/animationLibraryApi';
 
 export default function loadGamelab(options) {
   options.skinsModule = skins;
-  var gamelab = new GameLab();
+  return getDefaultListMetadata().then(defaultSprites => {
```
Comment (maddiedierker) [inline apps/src/sites/studio/pages/init/loadGamelab.js]: i'm concerned about the cascading implications of making these instantiations async. if `getDefaultListMetadata` fails, we currently don't instantiate gamelab/spritelab at all -- is that the behavior we want?

File: apps/src/sites/studio/pages/init/loadGamelab.js [hunk 4]
```diff
@@ -6,7 +6,7 @@ import levels from '@cdo/apps/p5lab/levels';
 
 export default function loadGamelab(options) {
   options.skinsModule = skins;
-  var gamelab = new GameLab();
```
Comment (epeach) [inline apps/src/sites/studio/pages/init/loadGamelab.js:9]: Gamelab doesn't use default animations, so we don't need to load default animations before constructing our GameLab

**Review by maddiedierker — APPROVED** [review]
great work, erin, and thank you for persisting on this -- it was much more complicated than we originally thought!

**Pull request merged**